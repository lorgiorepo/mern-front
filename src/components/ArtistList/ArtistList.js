import React, {useEffect} from 'react';
import {makeStyles} from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

import {useActions, useArtistsAPI} from '../../features/artist';

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
});


const ArtistList = () => {
  const classes = useStyles();

  const {getArtists} = useActions();
  
  useEffect(() => {
    getArtists();
  }, []);

  const {artists} = useArtistsAPI();
    return (
      <TableContainer component={Paper}>
        <Table className={classes.table} aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>Name</TableCell>
              <TableCell>Lastname</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {artists.map((artist, index) => (
              <TableRow key={artist.name}>
                <TableCell align="right">{artist.name}</TableCell>
                <TableCell align="right">{artist.lastname}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    );
}

export default ArtistList;