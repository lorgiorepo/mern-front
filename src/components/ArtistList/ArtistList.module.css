.lista {
    box-shadow: 0 4px 16px rgba(20,23,28,.25);
    padding: 0;
    border-radius: 7px;
    background: #282A36 !important; 
  }