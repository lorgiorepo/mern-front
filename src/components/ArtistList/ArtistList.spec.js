import React from 'react';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react';
import MockAdapter from 'axios-mock-adapter';
import axios from 'axios';
import promise from 'redux-promise-middleware';
import configureStore from 'redux-mock-store';
import ArtistList from './ArtistList';

describe('components > ArtistList', () => {

    const mockStore = configureStore([promise]);
    const mockAxios = new MockAdapter(axios);

    beforeEach(() => {
        mockAxios.resetHandlers();
    });

    it('renders artist list', () => {
        const store = mockStore({
            artistsData: {
                artists: []
            },
        });

        const {asFragment} = render(<ArtistList />, {
            wrapper: ({children}) => <Provider store={store}>{children}</Provider>,
        });

        expect(asFragment()).toMatchSnapshot();
    });
});