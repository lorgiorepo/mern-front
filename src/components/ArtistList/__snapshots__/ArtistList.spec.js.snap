// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > ArtistList renders artist list 1`] = `
<DocumentFragment>
  <div>
    <h1>
      All Artist
    </h1>
    <ul />
  </div>
</DocumentFragment>
`;
